recipes/gtest/all/conanfile.py

```diff
@@ -31,12 +31,19 @@ def _minimum_compilers_version(self):
             return {
                 "Visual Studio": "14"
             }
+        elif self.version == "1.10.0":
+            return {
+                "Visual Studio": "14",
+                "gcc": "4.8.1",
+                "clang": "3.3",
+                "apple-clang": "5.0"
```

Comment:
This does not work with the version compare, this will still force >=5

You'll need the `lazy_semver_lt` https://github.com/conan-io/conan-center-index/blob/f03c09a68e0159488602bf7d51d1aa4f11a0f590/recipes/sqlite_orm/all/conanfile.py#L33

Reply:
Could you expand a bit more for my understanding? `tools.Version("4.8.3") < "4.8.1"` yields false, while `tools.Version("4.8.0") < "4.8.1"` yields true so that seems to work as intended (running conan 1.33.1)?

Reply:
Excellent questions! 🤓 

Firstly the profiles used by CCI are **only** the compiler major. 
So our comparisons would be `tools.Version("4.8.3") < "4"`. Notice the difference in length?

There's a limitation (or feature) about the SemVer used underneath Conan which makes the assumption the comparison is only valid at the same "semantic level" ... so our comparison becomes `"4" < "4"` which is false!

In order to overcome this while still providing accurate recipes, a few community members came up with the "lazy semver" that I linked above.

I expect us to encounter the same problem here

Reply:
Ok little bit more clear. But isn't that valid? If compiler version is 4 you have no guarantee that gtest can compile because not any 4 version is valid. And it does seem to work for 4.9. Or do you think if it is 4 we should give a warning and still compile?

https://c3i.jfrog.io/c3i/misc/logs/pr/4664/3/gtest/1.10.0/2e54198abd8e5becf79ce4c6bc678ad47ec3ff1d/create_stdout.txt

Reply:
> But isn't that valid?

Yes absolutely. And the work you've done is spot on.

The challenge for CCI is the consumers expectations.

↪️ There should be a GCC version 4 package.

if they had a precise profile they would get more feedback but that's less common.
I believe most consumers just download the binaries and rarely build.

Reply:
Ah I see. So only the uploaded version has a truncated version. 

So in CCI CI you might build with 4.8.3 (so the profiles are still with compiler minor/patch) but as soon as they are submitted to bintray/artifactory/jfrog things (not sure what to call it :)) the version in the profile is truncated to major. If you then download that version it will give an error as 4 < 4 is false, even though it contains a working binary?

Reply:
> in CCI CI you might build with 4.8.3 [...] the profiles are still with compiler minor/patch

So you can check the logs above the CCI profiles have the minor but **no patch**... This is not changed by the upload JFrog platform preserves the same profile (otherwise it would break the `package_id` logic).

> the version in the profile is truncated

This is 💯 accurate and where the problem/challenge begins!

***

So I was a little bit wrong... GCC-4 seems to be different than all the other compilers (and versions) The default settings does not have **major only** as a valid option.

https://github.com/conan-io/conan/blob/10e7d5b1e534cc665fa4c403cb560e5de20364d6/conans/client/conf/__init__.py#L67

And openssl was compiled with `settings.compiler.version = 4.9` which is different than gcc-5 which was really compiled with 5.5 but that not in the `package_id` (it just says 5

- 4: openssl/1.1.1j/_/8e78761867c9a85805eaa65a157bcebf/2345d377da94e8ffae1333469573b8215c98c514/71a8978b20329a46daab0c28217b92c4
- 5: openssl/1.1.1j/_/8e78761867c9a85805eaa65a157bcebf/37dbc353875958e967097ad32a2b5aa8894ae5fb/23cdf0197a8625429df183251aa46b46

***

This should probably be documented 🤔

Reply:
Ok added the lazy semver, because even with 4.8/4.9 4.8.3 will still be seen as a bad package.